adding  to the stats

dbolser opened the pull request:
if you pass options via , they don't show up in the stats, which can be a problem for debugging

dbolser commented:
I'm pushing this against `release/112`, but not really sure that's the right thing to do...

I've tested this on 110, and it's working fine for me.

dglemos commented:
Thanks for the update, @dbolser!
The reason why the tests are failing is this one: https://github.com/Ensembl/ensembl-vep/blob/release/112/t/Stats.t#L409-L423
In the test the expected output doesn't include the new 'Full command line options'. This test needs to be updated to reflect the new behaviour.
Can you also please change the base to postreleasefix/112?

dbolser commented:
Thanks, I'll update the test. How to decide which branch to use when
creating a PR?

On Wed, Nov 22, 2023, 6:06 PM Diana Lemos ***@***.***> wrote:

> Thanks for the update, @dbolser <https://github.com/dbolser>!
> The reason why the tests are failing is this one:
> https://github.com/Ensembl/ensembl-vep/blob/release/112/t/Stats.t#L409-L423
> In the test the expected output doesn't include the new 'Full command line
> options'. This test needs to be updated to reflect the new behaviour.
> Can you also please change the base to postreleasefix/112?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Ensembl/ensembl-vep/pull/1560#issuecomment-1823244492>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA7NE7IXV4IZRB4XZ35P4LYFY5KJAVCNFSM6AAAAAA7TASSSCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMRTGI2DINBZGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

dglemos commented:
The branch is the latest postreleasefix which right now is postreleasefix/112.

dbolser commented:
Right, but is it the right branch for me to PR against?

Last time I did a PR against the default branch, then was asked to move it
to a prereleasebugfix/ (IIRC)... Is there a quick tutorial saying which
branch to PR against?


Many thanks,


On Thu, 23 Nov 2023 at 10:16, Diana Lemos ***@***.***> wrote:

> The branch is the latest postreleasefix which right now is
> postreleasefix/112.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Ensembl/ensembl-vep/pull/1560#issuecomment-1824125726>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA7NE536M7GZM64RW2CWHLYF4O6NAVCNFSM6AAAAAA7TASSSCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMRUGEZDKNZSGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

nuno-agostinho commented on modules/Bio/EnsEMBL/VEP/Stats.pm:
```diff
@@ -645,6 +645,7 @@ sub generate_run_stats {
   my @return = (
     ['Species', $self->species],
     ['Command line options', '<pre>'.join(" ", @opts).'</pre>'],
+    ['Full command line options', '<pre>'.$self->config->{_params}->{full_command}.'</pre>'],
```
Hey @dbolser,

Thanks for your PR.

It depends on our internal release schedule, but usually any PR should be merged into the latest `postreleasefix` branch.

I think in this case it would be better to simply replace the functionality of `Command line options`, so we return the same string as in the VCF output. What do you think?

```suggestion
    ['Command line options', '<pre>'.$self->config->{_params}->{full_command}.'</pre>'],
```

Let me know if you have any other questions. Thanks!

Best regards,
Nuno

nuno-agostinho replied:
Hi @dbolser, are you fine with me making the change that I suggested? Let me know, thanks!

dglemos commented:
> Right, but is it the right branch for me to PR against? Last time I did a PR against the default branch, then was asked to move it to a prereleasebugfix/ (IIRC)... Is there a quick tutorial saying which branch to PR against? Many thanks,
> […](#)
> On Thu, 23 Nov 2023 at 10:16, Diana Lemos ***@***.***> wrote: The branch is the latest postreleasefix which right now is postreleasefix/112. — Reply to this email directly, view it on GitHub <[#1560 (comment)](https://github.com/Ensembl/ensembl-vep/pull/1560#issuecomment-1824125726)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAA7NE536M7GZM64RW2CWHLYF4O6NAVCNFSM6AAAAAA7TASSSCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMRUGEZDKNZSGY> . You are receiving this because you were mentioned.Message ID: ***@***.***>

Hi @dbolser,
I apologise - I missed this message.
There is no strict rule on which branch to target for a pull request, it depends on the release you want to include the change in.
Currently, the live release is 113. If you'd like to include changes for the upcoming release, your PR should be against postreleasefix/114. This will eventually be merged into both release/114 and main. If the change is not urgent, you can target main, which will be part of the future release (in this case it wil be 115).

dglemos commented:
I've updated the base branch to postreleasefix/114. If still relevant, can you please update your PR accordingly?
Thanks

dbolser commented:
will do

On Fri, 28 Feb 2025 at 11:27, Diana Lemos ***@***.***> wrote:

> I've updated the base branch to postreleasefix/114. If still relevant, can
> you please update your PR accordingly?
> Thanks
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Ensembl/ensembl-vep/pull/1560#issuecomment-2690405772>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA7NE5BNIMAAVUPJJH35KL2SBBZXAVCNFSM6AAAAABYCCQLU6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJQGQYDKNZXGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> [image: dglemos]*dglemos* left a comment (Ensembl/ensembl-vep#1560)
> <https://github.com/Ensembl/ensembl-vep/pull/1560#issuecomment-2690405772>
>
> I've updated the base branch to postreleasefix/114. If still relevant, can
> you please update your PR accordingly?
> Thanks
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Ensembl/ensembl-vep/pull/1560#issuecomment-2690405772>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA7NE5BNIMAAVUPJJH35KL2SBBZXAVCNFSM6AAAAABYCCQLU6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMOJQGQYDKNZXGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
